pkg/deployment/utils/apply_utils.go

```diff
@@ -697,9 +697,19 @@ func (a *ApplyDeploymentsUtil) ApplyDeployments(deployments []*deployment.Deploy
 			sctx.Failed()
 		}()
 
+		var message *string
+
+		if d.Message != nil {
+			message = d.Message
+		}
+
+		if d.Config.Message != nil {
+			message = d.Config.Message
+		}
+
 		barrier := d.Config.Barrier || d.Barrier
 		if barrier {
-			sctx := status.StartWithOptions(a.ctx, status.WithStatus("Waiting on barrier..."), status.WithTotal(1))
+			sctx := status.StartWithOptions(a.ctx, status.WithStatus("Waiting on barrier: "+*message), status.WithTotal(1))
```

Comment:
Better to have something like:
```go
barrierMessage := "Waiting on barrier..."
if d.Config.Message != nil {
  barrierMessage = fmt.Sprintf("Waiting on barrier: %s", *d.Config.Message)
}
sctx := status.StartWithOptions(a.ctx, status.WithStatus(barrierMessage), status.WithTotal(1))
```

this way it also looks good when the message is empty.

Reply:
Used your proposed refactor also to prevent nil pointer exception